plugin/builtin/patchtransformer/PatchTransformer_test.go

```diff
@@ -198,6 +199,224 @@ Patch: "something"
+        name: nginx
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: yourDeploy
+spec:
+  template:
+    metadata:
+      labels:
+        new-label: new-value-with-multipleSMPatchesFile
+`
+
+const multipleSMPatchesSuccesfulResult = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    old-label: old-value
+  name: myDeploy
+spec:
+  replica: 2
+  template:
+    metadata:
+      labels:
+        old-label: old-value
+    spec:
+      containers:
+      - image: public.ecr.aws/nginx/nginx:mainline
+        name: nginx
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    new-label: new-value
+  name: yourDeploy
+spec:
+  replica: 1
+  template:
+    metadata:
+      labels:
+        new-label: new-value-with-multipleSMPatchesFile
+    spec:
+      containers:
+      - image: nginx:1.7.9
+        name: nginx
+---
+apiVersion: apps/v1
+kind: MyKind
+metadata:
+  label:
+    old-label: old-value
+  name: myDeploy
+spec:
+  template:
+    metadata:
+      labels:
+        old-label: old-value
+    spec:
+      containers:
+      - image: nginx
+        name: nginx
+`
+
+func TestMultipleSMPatchesWithFilePath(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarness(t).
+		PrepBuiltin("PatchTransformer")
+	defer th.Reset()
+
+	th.WriteF(`multiplepatches.yaml`, multipleSMPatchesFile)
+
+	th.RunTransformerAndCheckResult(`
+apiVersion: builtin
+kind: PatchTransformer
+metadata:
+  name: notImportantHere
+Path: multiplepatches.yaml
+`, someDeploymentResources, multipleSMPatchesSuccesfulResult)
+}
+
+func TestMultipleSMPatchesWithPatch(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarness(t).
+		PrepBuiltin("PatchTransformer")
+	defer th.Reset()
+
+	th.WriteF(`multiplepatches.yaml`, multipleSMPatchesFile)
+
+	th.RunTransformerAndCheckResult(`
+apiVersion: builtin
+kind: PatchTransformer
+metadata:
+  name: notImportantHere
+patch: |-
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: myDeploy
+  spec:
+    template:
+      spec:
+        containers:
+        - image: public.ecr.aws/nginx/nginx:mainline
+          name: nginx
+  ---
+  apiVersion: apps/v1
+  kind: Deployment
+  metadata:
+    name: yourDeploy
+  spec:
+    template:
+      metadata:
+        labels:
+          new-label: new-value-with-multipleSMPatchesFile
+`, someDeploymentResources, multipleSMPatchesSuccesfulResult)
+}
+
+func TestMultipleSMPatchesAndTarget(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarness(t).
+		PrepBuiltin("PatchTransformer")
+	defer th.Reset()
+
+	th.WriteF(`multiplepatches.yaml`, multipleSMPatchesFile)
+
+	th.RunTransformerAndCheckError(`
+apiVersion: builtin
+kind: PatchTransformer
+metadata:
+  name: notImportantHere
+Path: multiplepatches.yaml
+target:
+  name: .*Deploy
+  kind: Deployment
+`, someDeploymentResources, func(t *testing.T, err error) {
+		t.Helper()
+		if err == nil {
+			t.Fatalf("Multiple SMPatch will cause error if use 'target' field, but error didn't happen.")
+		}
```

Comment:
nit: This check is not necessary given the `require.ErrorContains` call below.

Sorry I misguided you in my previous comment: https://github.com/kubernetes-sigs/kustomize/pull/5194#discussion_r1287310373. In that comment I suggested that all test output needs to inform developers of the expected error message. I think what I meant to say is that the original "expected error" message sounded ambiguous. I think `require.ErrorContains` is enough because
1. It prints "An error is expected but got nil.", which prints the actual value and aligns with developer expectations because `require` is a standard library.
2. Given that `require.ErrorContains` checks for `err != nil`, we can shorten this test case by removing this extra check.

Totally my bad.

Reply:
Thanks, You are right. The `ErrorContains` line is enough to check the message.
I'll remove the `if err == nil ` check.

> Totally my bad.
Please don't care about it.